File: leet_GRDY_0455_cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        gg, ss = 0, 0
        g.sort()
        s.sort()
        while gg < len(g) and ss < len(s):
            if g[gg] <= s[ss]:
                gg += 1
            ss += 1
        return gg
        

File: leet_GRDY_0455_cookies.go
func findContentChildren(g []int, s []int) int {
    Solutions := []func( []int, []int ) int {
        Solution_2_pointers,
    }
    return Solutions[ rand.Intn( len(Solutions))] (g, s)
}

func Solution_2_pointers(g []int, s []int) int {

    fmt.Println("/Solution_2_pointers")
    sort.Ints(g)
    sort.Ints(s)
    L, R := 0, 0
    for L < len(g) && R < len(s) {
        if g[L] <= s[R] {
            L++
        }
        R++
    }
    return L
}
